packages/block-library/src/query-pagination/edit.js

```diff
@@ -23,5 +53,26 @@ export default function QueryPaginationEdit() {
 		],
 		orientation: 'horizontal',
 	} );
-	return <div { ...innerBlocksProps } />;
+	const context = useMemo( () => ( { paginationArrow } ), [
+		paginationArrow,
+	] );
+	return (
+		<>
+			{ hasNextPreviousBlocks && (
+				<InspectorControls>
+					<PanelBody title={ __( 'Settings' ) }>
+						<QueryPaginationArrowControls
+							value={ paginationArrow }
+							onChange={ ( value ) => {
+								setAttributes( { paginationArrow: value } );
+							} }
+						/>
+					</PanelBody>
+				</InspectorControls>
+			) }
+			<BlockContextProvider value={ context }>
```

Comment:
Is this one still needed? In my mind the "providesContext" config is mean to do it automatically?

Reply:
It's not and I removed it. Thanks! :)